src/utils/getRawFetchHeaders.ts

```diff
@@ -0,0 +1,38 @@
+import { getValueBySymbol } from './getValueBySymbol'
+
+type RawHeadersMap = Map<string, string>
+
+/**
+ * Returns raw headers from the given `Headers` instance.
+ * @example
+ * const headers = new Headers([
+ *   ['X-HeadeR-NamE', 'Value']
+ * ])
+ * getRawFetchHeaders(headers)
+ * // { 'X-HeadeR-NamE': 'Value' }
+ */
+export function getRawFetchHeaders(
+  headers: Headers
+): RawHeadersMap | undefined {
+  const headersList = getValueBySymbol<object>('headers list', headers)
+
+  if (!headersList) {
+    return
+  }
+
+  const headersMap = getValueBySymbol<
+    Map<string, { name: string; value: string }>
+  >('headers map', headersList)
```

Comment:
I've noticed the raw headers map is buried a bit deeper in the symbols:

```
Headers
 - Symbol(headers list)
   - Symbol(headers map)
```

Reply:
This seems to be specific to a Node.js version. This is where the raw headers map is for me on 18.14.0 but, apparently, it wasn't there on 18.8.0 used in the CI. I've updated the Node version on CI, let's see how it fares.

Reply:
I confirm that's due to the Node.js difference in versions:

- On 18.14.0, `headers map` is `Map<normalizedName, { name: rawName, value: string }>`
- On 18.8.0, the same `headers map` symbol is `Map<normalizedName, value>`. Looks like older versions of Node doesn't even expose the raw header name in any way.

Reply:
They do in the `Symbol(headers list).entires` (not the function `entries`)

Reply:
I fixed this by checking what values `headersMap` actually has. If it's not an object with a `name` property, return undefined.

Reply:
@mikicho, unfortunately, `.entries()` on `headers list` contains already normalized header names:

```js
const h = new Headers([['aBc, 'value']])
getValueBySymbol('headers list', h).entries()
// [ ['abc', 'value'] ]
```